AsyncDisplayKit/Layout/ASLayoutable.h

```diff
@@ -150,18 +150,52 @@ NS_ASSUME_NONNULL_BEGIN
 
 @end
 
+
 #pragma mark - ASLayoutableStyle
 
+extern NSString * const ASLayoutableStyleWidthProperty;
+extern NSString * const ASLayoutableStyleMinWidthProperty;
+extern NSString * const ASLayoutableStyleMaxWidthProperty;
+
+extern NSString * const ASLayoutableStyleHeightProperty;
+extern NSString * const ASLayoutableStyleMinHeightProperty;
+extern NSString * const ASLayoutableStyleMaxHeightProperty;
+
+extern NSString * const ASLayoutableStyleSpacingBeforeProperty;
+extern NSString * const ASLayoutableStyleSpacingAfterProperty;
+extern NSString * const ASLayoutableStyleFlexGrowProperty;
+extern NSString * const ASLayoutableStyleFlexShrinkProperty;
+extern NSString * const ASLayoutableStyleFlexBasisProperty;
+extern NSString * const ASLayoutableStyleAlignSelfProperty;
+extern NSString * const ASLayoutableStyleAscenderProperty;
+extern NSString * const ASLayoutableStyleDescenderProperty;
+
+extern NSString * const ASLayoutableStyleLayoutPositionProperty;
+
+@protocol ASLayoutableStyleDelegate <NSObject>
+- (void)style:(id)style propertyDidChange:(NSString *)propertyName;
+@end
+
 @interface ASLayoutableStyle : NSObject <ASStackLayoutable, ASStaticLayoutable>
 
+#pragma mark - Delegate
 
-#pragma mark - Sizing
+/**
+ * @abstract Initializes the layoutable style with a specified delegate
+ */
+- (instancetype)initWithDelegate:(id<ASLayoutableStyleDelegate>)delegate;
```

Comment:
Looking to the future a bit, it's likely that `delegate` will have to be `readwrite` because there are plans to reuse these style objects to save resources.